Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerTest.java

```diff
@@ -69,15 +80,22 @@
 /**
  * @author - johnny850807@gmail.com (Waterball)
  */
-@ActiveProfiles({Profiles.JWT, Profiles.EMBEDDED_MONGO})
+@Testcontainers
+@ActiveProfiles({Profiles.JWT, Profiles.EMBEDDED_MONGO, Profiles.REDIS})
 @AutoConfigureDataMongo
-@ContextConfiguration(classes = SpringBootProblemApplication.class)
+@ContextConfiguration(classes = {SpringBootProblemApplication.class, ProblemControllerTest.RedisConfig.class})
 public class ProblemControllerTest extends AbstractSpringBootTest {
 
+    public static final String REDIS_IMAGE_NAME = "redis";
+    public static final int REDIS_PORT = 6379;
     public static final int ADMIN_ID = 12345;
     public static final int STUDENT1_ID = 22;
     public static final String API_PREFIX = "/api/problems";
 
+    @Container
+    public static GenericContainer<?> REDIS = new GenericContainer<>(
+            DockerImageName.parse(REDIS_IMAGE_NAME)).withExposedPorts(REDIS_PORT);
+
```

Comment:
Nice one, thanks!

@ShawnSWu @timt1028 @snowmancc @Chaoyuuu Let's see how to run Redis container when running the Spring Boot Test (using the [TestContainers](https://www.testcontainers.org/) library)